packages/dds/tree/src/test/shared-tree/opSize.bench.ts

```diff
@@ -0,0 +1,1135 @@
+        children: [],
+    },
+};
+
+const childrenFieldKey: FieldKey = brand("children");
+
+/*
+ * Updates the given `tree` to the given `schema` and inserts `state` as its root.
+ */
+function initializeTestTree(tree: ISharedTree, state: JsonableTree = initialTestJsonTree) {
+    tree.storedSchema.update(fullSchemaData);
+    // inserts a node with the initial AppState as the root of the tree
+    tree.runTransaction((forest, editor) => {
+        const writeCursor = singleTextCursor(state);
+        const field = editor.sequenceField(undefined, rootFieldKeySymbol);
+        field.insert(0, writeCursor);
+        return TransactionResult.Apply;
+    });
+}
+
+const getJsonNode = (desiredByteSize: number): JsonableTree => {
+    const node = {
+        type: childSchema.name,
+        fields: {
+            data: [{ value: "", type: stringSchema.name }],
+        },
+    };
+
+    let nodeByteSize = new TextEncoder().encode(JSON.stringify(node)).length;
+
+    const sizeIncrementor = "a"; // 1 byte
+    const incrementorByteSize = new TextEncoder().encode(sizeIncrementor).length;
+
+    while (nodeByteSize < desiredByteSize) {
+        node.fields.data[0].value += sizeIncrementor;
+        nodeByteSize += incrementorByteSize;
+    }
+
+    return node;
+};
+
+const getChildrenlength = (tree: ISharedTree) => {
+    const cursor = tree.forest.allocateCursor();
+    moveToDetachedField(tree.forest, cursor);
+    cursor.enterNode(0);
+    cursor.enterField(childrenFieldKey);
+    const length = cursor.getFieldLength();
+    cursor.free();
+    return length;
+};
+
+const assertChildNodeCount = (tree: ISharedTree, nodeCount: number) => {
+    const cursor = tree.forest.allocateCursor();
+    moveToDetachedField(tree.forest, cursor);
+    cursor.enterNode(0);
+    cursor.enterField(childrenFieldKey);
+    assert.equal(cursor.getFieldLength(), nodeCount);
+    cursor.free();
+};
+
+const assertChildValuesEqualExpected = (
+    tree: ISharedTree,
+    editPayload: Value,
+    childCount: number,
+) => {
+    const cursor = tree.forest.allocateCursor();
+    moveToDetachedField(tree.forest, cursor);
+    cursor.enterNode(0);
+    cursor.enterField(childrenFieldKey);
+    cursor.enterNode(0);
+    assert.equal(cursor.value, editPayload);
+
+    let currChildCount = 1;
+    while (cursor.nextNode() && currChildCount < childCount) {
+        assert.equal(cursor.value, editPayload);
+        currChildCount++;
+    }
+    cursor.free();
+};
+
+// Creates a json tree with the desired number of children and the size of each child in bytes.
+const getInitialJsonTreeWithChildren = (numChildNodes: number, childNodeByteSize: number) => {
+    const childNode = getJsonNode(childNodeByteSize);
+    const jsonTree = {
+        type: parentSchema.name,
+        fields: {
+            children: [],
+        },
+    };
+    for (let i = 0; i < numChildNodes; i++) {
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore
```

Comment:
Why's this `@ts-ignore` necessary?

Reply:
The jsonable tree doesn't have a type definition so the `children` field is typed as `never[]` I could get around this by creating a type for the Jsonable tree if you think it makes sense.